src/System.Private.Reflection.Core/src/System/Reflection/Runtime/General/ThunkedApis.cs

```diff
@@ -58,7 +58,16 @@ public sealed override Stream GetManifestResourceStream(Type type, string name)
             return GetManifestResourceStream(sb.ToString());
         }
 
-        public sealed override string Location => string.Empty;
+        public sealed override string Location
+        {
+            get
+            {
+                if (AppContext.TryGetSwitch("Switch.System.Reflection.Assembly.SimulatedLocationInBaseDirectory", out bool isSimulated) && isSimulated)
+                    return Path.Combine(AppContext.BaseDirectory, GetName().Name + ".dll");
```

Comment:
Use the name of the manifest module? Then we don't have to speculate about the extension.

```suggestion
                    return Path.Combine(AppContext.BaseDirectory, ManifestModule.Name);
```